File: java/lasagna/src/main/java/Lasagna.java
public class Lasagna {

    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int elasped) {
        return expectedMinutesInOven() - elasped;
    }

    public int preparationTimeInMinutes(int layers) {
        return layers * 2;
    }

    public int totalTimeInMinutes(int layers, int time) {
        return preparationTimeInMinutes(layers) + time;
    }
}

File: elixir/lasagna/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven do
    40
  end

  def remaining_minutes_in_oven(minutes_in_oven) do
    expected_minutes_in_oven() - minutes_in_oven
  end

  def preparation_time_in_minutes(layers) do
    2 * layers
  end

  def total_time_in_minutes(layers, minutes_in_oven) do
    preparation_time_in_minutes(layers) +
      (expected_minutes_in_oven() - remaining_minutes_in_oven(minutes_in_oven))
  end

  def alarm do
    "Ding!"
  end
end
